platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/ProjectConfigAdapter.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.local;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+import org.eclipse.che.api.core.model.workspace.ProjectConfig;
+import org.eclipse.che.api.workspace.server.model.impl.ProjectConfigImpl;
+
+import java.lang.reflect.Type;
+
+/**
+ * The adapter which allows serialize deserialize modules in workspace.json. Each module in our model is presented as project and
+ * {@link ProjectConfig} describes module in our model.
+ *
+ * @author Dmitry Shnurenko
+ */
+public class ProjectConfigAdapter implements JsonDeserializer<ProjectConfig>, JsonSerializer<ProjectConfig> {
+
+    @Override
+    public ProjectConfig deserialize(JsonElement element, Type type, JsonDeserializationContext context) throws JsonParseException {
+        return context.deserialize(element, ProjectConfigImpl.class);
+    }
+
+    @Override
+    public JsonElement serialize(ProjectConfig moduleConfig, Type type, JsonSerializationContext context) {
+        return serializeModule(moduleConfig, context);
+    }
+
+    private JsonElement serializeModule(ProjectConfig moduleConfig, JsonSerializationContext context) {
+        JsonObject object = new JsonObject();
+
```

Comment:
Need to check that we need store to the Workspace. Look like not all attributes must be stored because some of them are calculate runtime.